assets/src/apps/delivery/layouts/deck/DeckLayoutView.tsx

```diff
@@ -42,13 +43,40 @@ const InjectedStyles: React.FC<{ css?: string }> = (props) => {
   return injected ? <style>{injected}</style> : null;
 };
 
+const getActualOwnerDetails = (expression: string, sequence: any) => {
+  const [activityId, part] = expression.split('|');
+  const activitySequence = sequence.find((entry: any) => entry?.custom?.sequenceId === activityId);
+  const activityLayerId = activitySequence?.custom?.layerRef;
```

Comment:
this could end up as undefined if it doesn't find a match in the sequence, and later when it's used you aren't checking that

Reply:
oh hmm, perhaps you are, but I think if it's undefined then you probably don't need to bother checking the value or anything at that point and either throw an error or just return the original